pupil-spa/src/app/check/check.component.ts

```diff
@@ -27,6 +27,14 @@ export class CheckComponent implements OnInit {
   public allowedStates: Array<string> = [];
   private totalNumberOfQuestions: number;
   public static readonly checkStateKey = 'checkstate';
+  private static warmupIntroRe = /^warmup-intro$/;
```

Comment:
Should we have a statics file for these?
Just so to keep purely the functional part for the service here.

Reply:
The state is internal to the component, so it makes sense to keep it together to me.